backend/src/tasks.ts

```diff
@@ -0,0 +1,349 @@
+    .addParam("nftAddress", "The address of the NFT contract to set")
+    .setAction(async (args, hre) => {
+        const quiz = await hre.ethers.getContractAt('Quiz', args.address);
+        await setNftAddress(hre, quiz, args.nftAddress);
+    });
+
+// Get list of valid coupons and spent coupons with the block number.
+task('getCoupons')
+    .addPositionalParam('address', 'contract address')
+    .setAction(async (args, hre) => {
+        await hre.run('compile');
+
+        console.log(`Coupons for quiz contract ${args.address}`);
+        const quiz = await hre.ethers.getContractAt('Quiz', args.address);
+
+        const [coupons, couponStatus] = await quiz.getCoupons();
+        let validCoupons: string[] = [];
+        let removedCoupons: string[] = [];
+        let spentCoupons = new Map<string, bigint>();
+        for (let i = 0; i < coupons.length; i++) {
+            if (couponStatus[i] == await quiz.COUPON_VALID()) {
+                validCoupons.push(coupons[i]);
+            } else if (couponStatus[i] == await quiz.COUPON_REMOVED()) {
+                removedCoupons.push(coupons[i]);
+            } else {
+                spentCoupons.set(coupons[i], couponStatus[i]);
+            }
+        }
+        let spentCouponsStr = ""
+        spentCoupons.forEach((value: bigint, key: string) => {
+            spentCouponsStr += `${key}:${value},`;
+        }
+        );
+        console.log(`Spent coupons (${spentCoupons.size}/${coupons.length}): ${spentCouponsStr.slice(0, spentCouponsStr.length - 1)}`);
+        console.log(`Valid coupons (${validCoupons.length}/${coupons.length}): ${validCoupons}`);
+        console.log(`Removed coupons (${removedCoupons.length}/${coupons.length}): ${removedCoupons}`);
+    });
+
+// Print out the Quiz status. Useful as a checklist.
+task('status')
+    .addPositionalParam('address', 'contract address')
+    .setAction(async (args, hre) => {
+        await hre.run('compile');
+
+        console.log(`Status of quiz contract ${args.address}`);
+        const quiz = await hre.ethers.getContractAt('Quiz', args.address);
+
+        // Questions.
+        const questions = await quiz.getQuestions("testCoupon1");
+        console.log(`Questions (counting from 0):`);
+        for (let i = 0; i < questions.length; i++) {
+            console.log(`  ${i}. ${questions[i].question}`);
+            for (let j = 0; j < questions[i].choices.length; j++) {
+                console.log(`     ${String.fromCharCode(97 + j)}) ${questions[i].choices[j]}`);
+            }
+        }
+
+        // Coupons.
+        try {
+            const coupons = await quiz.countCoupons();
+            console.log(`Coupons Available/All: ${coupons[0]}/${coupons[1]}`)
+        } catch (_) {
+        }
+
+        console.log(`Reward: ${hre.ethers.formatEther(await quiz.getTokenReward())} ROSE`)
+        console.log(`Payout Balance: ${hre.ethers.formatEther(await hre.ethers.provider.getBalance(await quiz.getAddress()))} ROSE`)
```

Comment:
```suggestion
        console.log(`Reward (in tokens): ${hre.ethers.formatEther(await quiz.getTokenReward())} ROSE`)
        console.log(`Payout Balance: ${hre.ethers.formatEther(await hre.ethers.provider.getBalance(await quiz.getAddress()))} ROSE`)
```

Add status for NFTs:
- NFT Reward Contract
  (shows nftAddress)
- NFT Metadata
  (fetches and shows nftAddress.tokenURIs[nftJsonHash])